File: 백준/알고리즘/구현/2161.py
card = int(input())

card_list = [card+1 for card in range(card)]
card_type = 0

result = []

while(len(card_list) > 1):
    if card_type == 0:
        result.append(card_list[0])
        card_type = 1
    elif card_type == 1:
        card_list.append(card_list[0])
        card_type = 0
    card_list.pop(0)

card_list = result + card_list
print(' '.join(list(map(str, card_list))))

File: 백준/자바스크립트/구현/2161.js
const fs = require("fs");
const path = process.platform === "linux" ? "/dev/stdin" : "../example.txt";
const input = fs.readFileSync(path).toString().trim().split("\n").map(Number);

const N = input[0];
const card = [];
for (let i = 1; i <= N; i++) card.push(i);

const result = [];
while (card.length > 1) {
  const a = card.shift();
  result.push(a);
  const b = card.shift();
  card.push(b);
}
result.push(card[0]);
console.log(result.join(" "));
